M2/Macaulay2/packages/Msolve.m2

```diff
@@ -0,0 +1,527 @@
+msolvePresent := run ("type msolve >/dev/null 2>&1") === 0
+-- msolveVersion := if msolvePresent then ... -- no way to get version info!!!
+msolveVersionNeeded := "0.6.4"
+msolvePresentAndModern := msolvePresent -- and match("^[0-9.]+$",msolveVersion) and msolveVersion >= msolveVersionNeeded
+
+newPackage(
+	"Msolve",
+	Version => "1.24.05", 
+    	Date => "July 2024",
+    	Authors => {{Name => "Martin Helmer", 
+		  Email => "mhelmer@ncsu.edu", 
+		  HomePage => "http://martin-helmer.com/"}, {Name => "Mike Stillman", 
+		  Email => "mike@math.cornell.edu", 
+		  HomePage => "https://math.cornell.edu/michael-e-stillman"},{Name => "Anton Leykin", 
+		  Email => "leykin@math.gatech.edu", 
+		  HomePage => "https://antonleykin.math.gatech.edu/"}},
+    	Headline => "An interface to the msolve package (https://msolve.lip6.fr/) which computes Groebner Basis and does real root isolation",
+    	AuxiliaryFiles => true,
+	DebuggingMode => false,       
+	OptionalComponentsPresent => msolvePresent
+	);
```

Comment:
This is missing a `Keywords` option, which is causing a test to fail.

Reply:
https://macaulay2.com/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/___Keywords.html says the option is optional... and provides no other guidance.

Reply:
It's on the [release checklist](https://github.com/Macaulay2/M2/wiki/Internals:-Release-Checklist) that all packages should have a keyword, so I added a test a while back that fails when a package doesn't.

The currently used keywords are the headings on the [packages provided with Macaulay2](https://www.macaulay2.com/doc/Macaulay2/share/doc/Macaulay2/Macaulay2Doc/html/_packages_spprovided_spwith_sp__Macaulay2.html) doc page.  We should probably improve the `Keywords` documentation to mention this -- I'll work on that.

I suppose this package would get "Groebner Basis Algorithms" and/or "Interfaces"?

Reply:
It is not a required input for `newPackage` because all inputs to `newPackage` are optional, even authors! But it is a requirement for being added to M2, so it should be documented somewhere on GitHub perhaps.